packages/react-theming/src/examples/compose/index.stories.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+import { initalize } from './../..';
+import { ThemeProvider } from './../../components/ThemeProvider/ThemeProvider';
+import { compose } from './../../compose';
+import { Variant } from './../../variant';
+import { ITheme, IColorRamp } from '../../theme.types';
+
+initalize();
+
+export default {
+  component: 'compose',
+  title: 'Compose Demos',
+};
+
+const emptyRamp: IColorRamp = { values: [], index: -1 };
+const theme: ITheme = {
+  components: {},
+  colors: {
+    background: 'white',
+    bodyText: 'black',
+    subText: 'black',
+    disabledText: 'green',
+    brand: emptyRamp,
+    accent: emptyRamp,
+    neutral: emptyRamp,
+    success: emptyRamp,
+    warning: emptyRamp,
+    danger: emptyRamp,
+  },
+  fonts: {
+    default: '',
+    userContent: '',
+    mono: '',
+  },
+  fontSizes: {
+    base: 1,
+    scale: 1,
+    unit: 'rem',
+  },
+  animations: {
+    fadeIn: {},
+    fadeOut: {},
+  },
+  direction: 'ltr',
+  spacing: {
+    base: 0,
+    scale: 0,
+    unit: 'rem',
+  },
+  radius: {
+    base: 0,
+    scale: 0,
+    unit: 'rem',
+  },
+  icons: {},
+  schemes: {},
+};
+
+const BaseDiv: React.FunctionComponent<{ classes: any }> = props => {
+  return <div className={props.classes.root}>{props.children}</div>;
+};
+
+const ComposedDiv = compose(BaseDiv as any, {
+  tokens: {
+    fontWeight: 300,
+    borderRadius: 0,
+  },
+  styles: (tokens: any) => {
+    return {
+      root: {
+        background: 'red',
+        fontWeight: tokens.fontWeight,
+        borderRadius: tokens.borderRadius,
+        margin: 10,
+        padding: 10,
+      },
+    };
+  },
+  variants: {
```

Comment:
I can see that this is inspired from the work we have done on https://github.com/microsoft/fluent-ui-react/pull/2200 however after playing a bit with it, I have some doubts about how it would work when you will need to combine variants together. More over, usually for some variants like disable, you want to override everything that was previously defined (like no hover styles for example), would lieke to hear from you how you envision these to work. Generally, I think it would be useful to see more complex styles examples that will require the above mention scenarios.

Reply:
Added another scenario with a `disabled` prop. The styles implementation isn't ideal (yet), but shows what is possible.